### ISSUE TITLE
request: docs for nescaline, freeboy and SID

### ISSUE THREAD

**Issue by opensourcefree** [LMMS/lmms#6532]
Hi

As you can read here I try to make a NES chiptune. (and later freeboy and SID).

https://lmms.io/forum/viewtopic.php?t=35767

It's difficult for 2 reasons:

1. There seems to be no documentation for nescaline, freeboy and SID which makes it very difficult for beginners.
What are the buttons doing, what channel represents a specific channel on the NES, ...
(See https://lmms.io/forum/viewtopic.php?t=35767 )

https://docs.lmms.io/user-manual/5-buil ... .3-freeboy

https://docs.lmms.io/user-manual/5-buil ... -nescaline

https://docs.lmms.io/user-manual/5-buil ... s/5.15-sid

2. What number of channels did the original hardware have and how are they mapped to the instrument channels?
Without documentation, I can't figure it out.

Is there somebody interested in writing chiptune tutorials for nescaline, freeboy and SID with specific focus on the different channels and a summary of what each button of the instrument is doing exactly? Maybe with example (small) songs for LMMS that we can download.

I also want to ask the LMMS documentation team for writing documentation on the LMMS documentation website for nesclaine, freebody and SID with similar content as above.

Thanks
Lisa

**Comment by opensourcefree** [comment]
> How the original NES worked has nothing to do with LMMS. Nescaline is just a NES-like instrument, it's not an exact copy of the NES sound chip. 

Is the same true for freeboy and SID? So are they instruments not exactly behaving as their original hardware and may have controls that were not on the original gameboy/C64?

**Comment by allejok96** [comment]
> So the goal of nescaline/freeboy/SID is to "sound" like the original sound hardware but not to clone it all the way

Yes. It's called fakebit. Nowdays we don't have the same technical limitations as they had in the 80s, so why enforce it on people creating music? 

There is more to it than just the plugin... If you want to make an authentic NES melody, you can't play more than 4 notes at the same time *ever*. To create authentic chip tune you must learn how the actual chip works.

**Comment by opensourcefree** [comment]
Can I maybe request a small option in nescaline that, when turned on, enforces nescaline to only allow hardware accurate controls?
So you can turn it on/off.
I don't know if this fits into LMMS' goals :)

**Comment by Monospace-V** [comment]
> Can I maybe request a small option in nescaline that, when turned on, enforces nescaline to only allow hardware accurate controls? So you can turn this option on/off. I don't know if this fits into LMMS' goals :)

I mean, it could be done, but usually any user interested in doing this would want to know the limitations themselves, and would by extension be capable of limiting themselves.
It sounds good, though, and I'd use it if it were there more often than I'd challenge myself, all right.
But four-note forced polyphony, beyond which you can't place notes? Is it feasible? Does this extend to release time, halfway through note, etc?
I don't know enough about what exactly is meant by limited polyphony on old devices to answer that.
Is it possible to put these into LMMS?
The LB302 plays a random note if two notes start at the same time. Do we want something like this or a hard limit on what you can place?

**Comment by opensourcefree** [comment]
I think maybe it's too much work just for this feature :)
As a non-developer I maybe think time could be used much better for other features.
I didn't realize this when I wrote my idea here.

**Comment by allejok96** [comment]
Well, a "limited" mode is maybe not that important, but documentation is. As a non-dev that's perhaps one area you can help with? If you want to and feel comfortable that is. Good bug reports are also a way of helping :)
